examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,247 @@
+# # Shallow Water Example: an unstable Bickley jet 
+#
+# This example shows how to use `Oceananigans.ShallowWaterModel` to simulate
+# the evolution of an unstable, geostrophically balanced, Bickley jet.
+# The model solves the governing equations for the shallow water model in
+# conservation form.  The geometry is that of a periodic channel
+# in the ``x``-direction with a flat bottom and a free-surface.  The initial
+# conditions are that of a Bickley jet with small amplitude perturbations.
+# The interested reader can see ["The nonlinear evolution of barotropically unstable jets," J. Phys. Oceanogr. (2003)](https://doi.org/10.1175/1520-0485(2003)033<2173:TNEOBU>2.0.CO;2)
+# for more details on this specific problem. 
+#
+# Unlike the other models, the fields that are simulated are the mass transports, 
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, LinearAlgebra, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the wavenumber that yields the most unstable
+# mode is ``k=0.95``. See [Linear-Stability-Calculators](https://github.com/francispoulin/Linear-Stability-Calculators/tree/main/ShallowWater/Julia/Cartesian)
```

Comment:
I'm skeptical to refer to a GH repo that is not a package. Will this repo be maintained etc? Is it clear to all users what this repo does?

Perhaps just stating that for these parameters the most unstable mode is a ``k = 0.95`` is enough?

Btw, I guess that for `Lx = 2π` the most unstable mode is at `k = 0.95`, right? And therefore, you expanded the domain a bit to `2π / 0.95` so that the most unstable mode fits, right? But we could simply leave `Lx = 2π` and then the `k = 1.0` mode will grow (with a slightly different growth rate). What's the problem doing so? Then we don't need explain that maximum instability is at `k = 0.95` and what not...

Reply:
This is a repo that I have created recently and have not turned into a package yet.  I agree it's probably simplest to not site it.  When I make it into a package letter, we can revisit this issue as to whether this would be helpful or just too much information.

I agree that simply pickin the domain to be `2\pi` is simpler and the way to go. I have included these changes.